lambdas/handlers/login_redirect_handler.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import os
+import time
+
+import boto3
+from botocore.exceptions import ClientError
+from oauthlib.oauth2 import WebApplicationClient, InsecureTransportError
+
+from utils.get_aws_region import get_aws_region
+from utils.lambda_response import ApiGatewayResponse
+from services.dynamo_service import DynamoDBService
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def lambda_handler(event, context):
+    return prepare_redirect_response(WebApplicationClient)
+
+
+def prepare_redirect_response(web_application_client_class):
+    try:
+        ssm_response = get_ssm_parameters()
+
+        oidc_parameters = {
+            parameter["Name"]: parameter["Value"]
+            for parameter in ssm_response["Parameters"]
+        }
+
+        oidc_client = web_application_client_class(
+            client_id=oidc_parameters["OIDC_CLIENT_ID"],
+        )
+
+        url, headers, body = oidc_client.prepare_authorization_request(
+            authorization_url=oidc_parameters["OIDC_AUTHORISE_URL"],
+            redirect_url=os.environ["OIDC_CALLBACK_URL"],
+            scope=["openid", "profile", "nationalrbacaccess", "associatedorgs"],
+        )
+
+        save_state_in_dynamo_db(oidc_client.state)
+
+        location_header = {"Location": url}
+
+    except ClientError as e:
+        logger.error(f"Error getting using aws client: {e}")
+        return ApiGatewayResponse(
+            500, "Server error", "GET"
+        ).create_api_gateway_response()
+    except InsecureTransportError as e:
+        logger.error(f"Error preparing auth request: {e}")
+        return ApiGatewayResponse(
+            500, "Server error", "GET"
+        ).create_api_gateway_response()
+    return ApiGatewayResponse(302, "", "GET").create_api_gateway_response(
+        headers=location_header
+    )
+
+
+def save_state_in_dynamo_db(state):
+    dynamodb_name = os.environ["AUTH_DYNAMODB_NAME"]
```

Comment:
May need a check on this env variable

Reply:
Will be done as part of the decorator work